### PR TITLE
Refactor federation spec.uri to spec.uriSecret

### PR DESCRIPTION
**Pull request by ChunyiLyu**
This closes #

**Note to reviewers:** remember to look at the commits in this PR and consider if they can be squashed
**Note to contributors:** remember to re-generate client set if there are any API changes

## Summary Of Changes

- federation upstream uri contains credentials and should not be saved in a CR manifest directly
- controller does not watch the uriSecret for now; this will be addressed in a separate issue and PR that will add same function to schema replication CR as well (issue: #126).

### CODE REVIEW

File: config/crd/bases/rabbitmq.com_federations.yaml [hunk 1]
```diff
@@ -74,17 +74,24 @@ spec:
...
-                description: The AMQP URI(s) for the upstream. Required property.
-                type: string
+              uriSecret:
+                description: Secret contains the AMQP URI(s) for the upstream. The
+                  Secret must contain the key `uri` or operator will error. Required
+                  property.
+                properties:
+                  name:
+                    description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+                      TODO: Add other useful fields. apiVersion, kind, uid?'
```
Comment (MirahImage) [inline config/crd/bases/rabbitmq.com_federations.yaml:84]: Is the `TODO` still meant to be here?

Comment (ChunyiLyu) [reply config/crd/bases/rabbitmq.com_federations.yaml:84]: It comes from kubernetes api corev1.LocalObjectReference directly: https://github.com/kubernetes/api/blob/b0d9a0e45e6f6f7c5f18d76d950a4d19e41d1f50/core/v1/types.go#L5407

File: system_tests/federation_system_test.go [hunk 2]
```diff
@@ -51,7 +64,7 @@ var _ = Describe("federation", func() {
 
 		Expect(upstream.Name).To(Equal(federation.Spec.Name))
 		Expect(upstream.Vhost).To(Equal(federation.Spec.Vhost))
-		Expect(upstream.Definition.Uri).To(Equal(federation.Spec.Uri))
+		Expect(upstream.Definition.Uri).To(Equal(federationUri))
```
Comment (coro) [inline system_tests/federation_system_test.go:67]: I'm surprised this passes, given one is a string and the other a `corev1.Secret`. Am I missing something?

Comment (coro) [reply system_tests/federation_system_test.go:67]: Ah I see, it's a naming collision with the string in line 21, being redefined in line 25. Could you rename the secret please?

**Review by coro — CHANGES_REQUESTED** [review]

File: system_tests/federation_system_test.go (hunk 2 above)
Comment (ChunyiLyu) [reply system_tests/federation_system_test.go:67]: Updated the name of the secret.

**Pull request merged**